client/src/api/providers.ts

```diff
@@ -0,0 +1,34 @@
+import { errors } from '../util/errors';
+
+type AliasedProviders = 'production' | 'staging';
+type ProviderBuilder = { main: string; workspaces: string };
+export type ApiProviders = AliasedProviders | ProviderBuilder;
```

Comment:
User can provide either their own provider builder (two different urls) or use ``production`` (default).

For development (we) will use ``staging``.